src/main/java/com/atlassian/db/replica/internal/SqlQuery.java

```diff
@@ -13,14 +13,18 @@ public SqlQuery(String sql) {
     }
 
     public boolean isWriteOperation(SqlFunction sqlFunction) {
-        return sqlFunction.isFunctionCall(sql) || isUpdate() || isDelete();
+        return sqlFunction.isFunctionCall(sql) || isUpdate() || isDelete() || isInsert();
     }
 
     public boolean isSelectForUpdate() {
         final String trimmedQuery = trimForSelectForUpdateCheck();
         return trimmedQuery.contains("for update") || trimmedQuery.contains("FOR UPDATE");
     }
 
+    public boolean isInsert() {
+        return sql.startsWith("insert") || sql.startsWith("INSERT");
```

Comment:
`sql.contains("insert")`

Reply:
Most db-replicas decisions are heuristics focused on performance over accuracy. We update the rules when we see it doesn't work on prod. String#contains is O(nm) (for the Java implementations I've seen. It's possible to implement it better O(n), but it's still costly), and startsWith is O(1) and passes the case from prod.

Reply:
`Most db-replicas decisions are heuristics focused on performance over accuracy. `

👍  Ok, lets follow this